iModelCore/ECDb/ECDb/ECSql/ValueExp.cpp

```diff
@@ -807,6 +807,21 @@ Exp::FinalizeParseStatus FunctionCallExp::_FinalizeParsing(ECSqlParseContext& ct
     if (m_isStandardSetFunction && m_functionName.EqualsI("count"))
         return FinalizeParseStatus::Completed;
 
+    //validation for InVirtualSet function - first argument for InVirtualSet should be VirtualSet or NULL
+    if (m_functionName.EqualsI("InVirtualSet"))
+        {
+        ValueExp* argExp = GetChildP<ValueExp>(0);
+        ECSqlTypeInfo::Kind typeKind = argExp->GetTypeInfo().GetKind();
+        if (typeKind != ECSqlTypeInfo::Kind::Unset && !argExp->GetTypeInfo().IsNull())
+            {
+            ctx.Issues().ReportV(IssueSeverity::Error, IssueCategory::BusinessProperties, IssueType::ECSQL, "Function '%s' can only be called with unset argument type. First argument is not unset.",

```

Comment:
Since this check is very specific to InVirtualSet can we make the error description more helpful for a user who has used InVirtualSet incorrectly?